site/app/templates/forum/ThreadPostForm.twig

```diff
@@ -198,6 +198,26 @@
         $("input[name='title']").change(function () {
             $(this).val($.trim($(this).val()));
         });
+
+        $(".thread-announcement-checkbox").click(function (){
+            if($(".thread-announcement-checkbox").prop("checked")){
+                $(".thread-announcement-checkbox").prop("checked", true);
+                $(".pinThread-checkbox").prop("checked", true);
+                $("#thread_status").val(0);
+            } else {
+                $(".thread-announcement-checkbox").prop("checked", false);
+                $(".pinThread-checkbox").prop("checked", false);
```

Comment:
make sure to stop it from being disabled here

you should just be able to add this line after `$(".pinThread-checkbox").prop('disabled', false);`